src/middlewares/check_auth.rs

```diff
@@ -117,11 +118,15 @@ where
 
         let token = token.to_string();
 
-        let components = request.app_data::<Data<AppComponents>>().unwrap().clone();
         let svc = self.service.clone();
-
         Box::pin(async move {
-            let user_id = get_user_id_from_token(components.into_inner(), &token).await;
+            let components = request.app_data::<Data<AppComponents>>().unwrap().clone();
```

Comment:
I know this line was like that from before 🫠, but is there any way to handle the `unwrap()`?